### PR TITLE
emscripten head: fix build

### PR DESCRIPTION
**Pull request by umireon**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The HEAD build of emscripten was broken because the fastcomp installation of the formula depended on the configure script disabled now by the upstream (LLVM).

The configure script suggests the recommended way to build is using CMake.
(https://github.com/llvm-mirror/llvm/blob/master/configure)

Thus, the build process of fastcomp when installing the HEAD version is changed to use CMake instead of the disabled configure script.

Note that the part for the stable version is kept unchanged, since the current stable version of fastcomp can still be built with the configure script.

### CODE REVIEW

File: Formula/emscripten.rb [hunk 1]
```diff
@@ -60,16 +62,32 @@ def install
...
+      "-DCMAKE_INSTALL_PREFIX=#{libexec}/llvm",
+      "-DCMAKE_BUILD_TYPE=Release",
+      "-DLLVM_TARGETS_TO_BUILD=X86;JSBackend",
+      "-DLLVM_INCLUDE_EXAMPLES=OFF",
+      "-DLLVM_INCLUDE_TESTS=OFF",
+      "-DCLANG_INCLUDE_EXAMPLES=OFF",
+      "-DCLANG_INCLUDE_TESTS=OFF",
+      "-DOCAMLFIND=/usr/bin/false",
+      "-DGO_EXECUTABLE=/usr/bin/false",
+    ]
```
Comment (umireon) [inline Formula/emscripten.rb]: Before inlining, I noticed `std_cmake_args` is defined for CMake builds in Homebrew.
I checked it and found that can be applicable for this fastcomp compilation.
Can I include this standard argument (and thus persist on the array) ?

Comment (umireon) [inline Formula/emscripten.rb]: ...like this:

```
    cmake_args = std_cmake_args.reject{|s| s["CMAKE_INSTALL_PREFIX"] }
    cmake_args += [
      "-DCMAKE_INSTALL_PREFIX=#{libexec}/llvm",
      "-DLLVM_TARGETS_TO_BUILD=X86;JSBackend",
      "-DLLVM_INCLUDE_EXAMPLES=OFF",
      "-DLLVM_INCLUDE_TESTS=OFF",
      "-DCLANG_INCLUDE_EXAMPLES=OFF",
      "-DCLANG_INCLUDE_TESTS=OFF",
      "-DOCAMLFIND=/usr/bin/false",
      "-DGO_EXECUTABLE=/usr/bin/false",
    ]
```

**Comment by MikeMcQuaid** [comment]
Some style violations but otherwise looking good: https://bot.brew.sh/job/Homebrew%20Core/10762/version=any_sierra/testReport/junit/brew-test-bot/sierra/style_emscripten/

**Comment by umireon** [comment]
I apologize for forgetting to run the style checker.
The updated formula is now passing the checker.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `emscripten: fix HEAD build`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request merged**